static_frame/core/util.py

```diff
@@ -3186,5 +3187,43 @@ def iloc_to_insertion_iloc(key: int, size: int) -> int:
         raise IndexError(f'index {key} out of range for length {size} container.')
     return key % size
 
+def get_unsized_children_iter(obj):
+    # Check if iterable or a string first for fewer isinstance calls on common types
+    if hasattr(obj, '__iter__') and not isinstance(obj, str):
+        if (
+            (isinstance(obj, np.ndarray) and obj.dtype == np.object)
+            or isinstance(obj, tuple)
+            or isinstance(obj, list)
+            or isinstance(obj, set)
+            or isinstance(obj, frozenset)
+        ):
+            yield from obj
+        elif isinstance(obj, dict):
+            yield from chain.from_iterable(obj.items())
+        else:
+            # The full size of the object is included in a getsizeof call
+            # e.g. FrozenAutoMap, integer numpy arrays, int, float, etc.
+            pass
 
-
+def all_nested_elements(obj: any, *, seen=None) -> tp.Iterable[any]:
+    seen = set() if seen is None else seen
+    if id(obj) in seen:
+        return
+    seen.add(id(obj))
+    for el in get_unsized_children_iter(obj):
+        yield from all_nested_elements(el, seen=seen)
+    yield obj
+
+def getsizeof_recursive(obj: any, *, seen=None) -> int:
+    seen = set() if seen is None else seen
+    return sum(getsizeof(el) for el in all_nested_elements(obj, seen=seen))
+
+def sizeof_helper(cls, self, *, seen):
```

Comment:
Wanted to just sum each slot element with something like this but I'm not sure it's possible for python to know the MRO.
```
def collect_slots(cls, self):
    if hasattr(cls, '__slots__'):
        yield from collect_slots(super(cls, self), self)
        yield from cls.__slots__
```

Reply:
```python
def collect_slots(obj):
    return frozenset().union(cls.__slots__ for cls in obj.__class__.__mro__)
```